src/test/java/net/snowflake/client/jdbc/HeartbeatAsyncLatestIT.java

```diff
@@ -83,4 +83,21 @@ public void testAsynchronousQuerySuccess() throws Exception {
   public void testAsynchronousQueryFailure() throws Exception {
     testFailure();
   }
+
+  /** Test that isValid() function returns false when session is expired */
+  @Test
+  @ConditionalIgnoreRule.ConditionalIgnore(condition = RunningOnGithubAction.class)
+  public void testIsValidWithInvalidSession() throws Exception {
+    Connection connection = null;
+    try {
+      connection = getConnection();
+      // assert that connection starts out valid
+      assertTrue(connection.isValid(5));
+      Thread.sleep(61000); // sleep 61 seconds to await session expiration time
```

Comment:
QQ: Is this because we set the session timeout to 60s somewhere else in `HeartbeatAsyncLatestIT`?

Reply:
Yes! To clarify, part of the test setup in HeartbeatAsyncLatestIT is to set the session expiration to 60s and the token expiration to 20s.